evm/vm/base.py

```diff
@@ -145,26 +143,42 @@ def import_block(self, block):
         )
 
         # run all of the transactions.
-        for transaction in block.transactions:
+        execution_data = [
             self.apply_transaction(transaction)
+            for transaction
+            in block.transactions
+        ]
```

Comment:
Something about generating a list this way makes it less obvious that `apply_transaction()` is also mutating the state in the VM. I guess if we become super consistent about `apply` => "mutation", then it's not a problem.

Reply:
Agreed.  Fixing in subsequent refactoring steps.